week4/trie/thalesmg/Main.hs

```diff
@@ -0,0 +1,56 @@
+import qualified Data.Map.Lazy as M
+
+data Trie a = Empty | Node (M.Map a (Trie a)) deriving (Show)
+
+mkTrie :: [a] -> Trie a
```

Comment:
This isn't a good name; `mkTrie` sounds like a function for creating a trie out of a bunch of strings, but it's not. (By the way, it'd be nice for a “function for creating a trie out of a bunch of strings” to exist as well and be top-level.)